geoh5py/shared/concatenation/drillholes_group_table.py

```diff
@@ -0,0 +1,377 @@
+        """
+        Pad the arrays in the list to match the size of the first array.
+
+        :param arrays: The list of arrays to pad.
+        :param ndv: The No Data Value of the data.
+
+        :return: The padded arrays.
+        """
+        # Pad other arrays in the list to match the size of the first array
+        padded_arrays = [arrays[0]]  # First array remains the same
+        for idx, array in enumerate(arrays[1:]):
+            pad_size = arrays[0].shape[0] - array.shape[0]
+            if pad_size > 0:
+                padded_array = np.pad(
+                    array,
+                    (0, pad_size),
+                    mode="constant",
+                    constant_values=(ndv[idx], ndv[idx]),
+                )
+            else:
+                padded_array = array
+            padded_arrays.append(padded_array)
+
+        return padded_arrays
+
+    def add_values_to_property_group(
+        self,
+        name: str,
+        values: np.ndarray,
+    ):
+        """
+        Push the values to each drillhole of the property group based on association.
+
+        :param name: The name of the data to push.
+        :param values: The values to push.
+        """
+        if not isinstance(name, str) or name in self.parent.data:
+            raise KeyError("The name must be a string not present in data.")
+
+        # ensure the length of the values is the same as the length of the template
+        if values.shape != self.parent.data[self.association[0]].shape:
+            raise ValueError(
+                "The length of the values must be the same as the association "
+                f"({self.parent.data[self.association[0]].shape})."
+            )
+
+        for drillhole_uid, indices in self.index_by_drillhole.items():
+            # get the drillhole
+            drillhole: ConcatenatedDrillhole = self.parent.workspace.get_entity(  # type: ignore
+                UUID(drillhole_uid.decode("utf-8"))
+            )[
+                0
+            ]
+
+            # define the associations values
+            drillhole_association = []
+            for key in self.association:
+                drillhole_association.append(
+                    self.parent.data[key][
+                        indices[key][0] : indices[key][0] + indices[key][1]
+                    ]
+                )
+
+            # add data to the drillhole
+            drillhole.add_data(
+                {
+                    name: {
+                        "values": values[
+                            indices[self.association[0]][0] : indices[
+                                self.association[0]
+                            ][0]
+                            + indices[self.association[0]][1]
+                        ],
+                        self.association_type: np.array(drillhole_association).T,
+                    },
+                },
+                property_group=self.name,
```

Comment:
I think this would create a new property group?
Simpler is just append to the current one like this

```suggestion
                property_group=self._property_groups[drillhole.uid],
```

Reply:
No, as data in drillhole Concatenator, property Groups are based on names. 
GA recognize it